gap/prop.gi

```diff
@@ -589,3 +589,48 @@ end);
 
 InstallMethod(IsHamiltonianDigraph, "for a digraph with hamiltonian path",
 [IsDigraph and HasHamiltonianPath], x -> HamiltonianPath(x) <> fail);
+
+InstallMethod(IsDigraphCore, "for a digraph",
+[IsDigraph],
+function(D)
+  local hook, proper_endo_found, N;
+
+  N := DigraphNrVertices(D);
+  if DigraphHasLoops(D) and N > 1 then
+    return false;
+  elif IsCompleteDigraph(D) then
+    return true;
+  elif IsBipartiteDigraph(D) and N > 2 then
+    return false;
+  fi;
+  # The core of a digraph with loops is a vertex with a loop, and the core of a
+  # bipartite digraph is the complete digraph on 2 vertices.
```

Comment:
A non-empty bipartite digraph. If there’s no edges the core is the empty digraph with one vertex.